### ISSUE TITLE
Ideas for the new version of GNS3

### ISSUE THREAD

**Issue by uktricky** [GNS3/gns3-gui#1055]
**Grouping of devices .. to allow start or stop of a particular group**
I often create "sites" (groups of devices) with a number of devices and then link to other "sites".  Would be useful to have a "start devices in Group x" button - I know I can outline the devices and right-click on one of them and do a Start which is fine if your layouts are neat enough and devices don't cross each other.  Start "Icon" could then have "All" at the top and groups listed below with a status (Red/Green) to show if running.  All devices could default to group 1?

**Potentially collapsing a group of devices into a Building Image**
As per the above.  Ability to minimise a group of devices into a single icon with links coming out to link to other devices.

**Tabbed view in GNS3**
Move groups of devices between tabs again allowing an off-page connector to link tabs together and perhaps have a text editor (notepad) on a different tab(s)

**Be able to sort devices in the Topology Summary**

**Be able to highlight a device on the Topology Summary and Centre the "Image" (Topology) on this device

Show a count of certain devices/images in use - Purely for reference**

**Be able to add "Random Issues" to aid Troubleshooting skills **
Slow links due to errors (CRC's etc)
Potentially a flapping link causing neighbor issues / reconvergence
Loop issues
Black hole (one way traffic)
etc
etc

**Comment by jean-christophe-manciot** [comment]
Yes: a **multi-tier topology,** definitely a very good idea, for example multiple DCs linked together with DCIs; when double-clicking a DC node, we can enter that particular DC topology which was hidden in the previous view.

**Comment by danieldemers** [comment]
My 2 cents.

Be able to group devices together so you can hide parts of the topology you do not care about at the moment. The devices are still running, this is simply for view only. 

Be able to tag network elements (IOU, Appliances, associated network links) and group them together. For instance if I have several overlay or underlay designs that are not interacting with each other and out of 15 devices each touches 7 of them with some overlap. It be great to tag those 7 devices for say DMVPN, VLAN, routing domain, security zone, etc. and in a tabbed/new view only shows the devices I care about at the moment.

Keep the ability to show all but if I am only working on an MPLS lab then only show the MPLS cloud with the PE/CEs.

This would be a manual process for the designer of the lab, however it helps with creating a scope of devices you care about depending on the work you are doing at that moment.

AND/OR

Extract the Configuration from Cisco/Juniper devices and create an underlay/overlay view. I only want to see where VLAN 10 is trunked and access. This may need to be a mnaual process but that would be helpful also.

**Comment by noplay** [comment]
I think create tag is the best way to do that. So third parties tools (or user) could generate the tag based on what is is inside the OS or whatever is relevant.

**Comment by grossmj** [comment]
Tag support has been added since v3.1.0
